File: packages/core/src/schemas/prd.ts
import { z } from 'zod';
import { entityIdSchema, entityRefSchema, statusSchema } from './common.js';

export const prdFrontmatterSchema = z.object({
  type: z.literal('prd'),
  id: entityIdSchema,
  title: z.string().min(1),
  status: statusSchema,
  owner: z.string().nullable().optional(),
  epic_refs: z.array(entityRefSchema).default([]),
  created_at: z.string().optional(),
  updated_at: z.string().optional(),
});

export const prdSchema = prdFrontmatterSchema.extend({
  body: z.string().default(''),
  filePath: z.string(),
});

export type PrdFrontmatter = z.infer<typeof prdFrontmatterSchema>;
export type Prd = z.infer<typeof prdSchema>;

File: go/internal/schema/prd.go
package schema

import "time"

type PrdFrontmatter struct {
	Type      EntityType  `yaml:"type" json:"type"`
	ID        string      `yaml:"id" json:"id"`
	Title     string      `yaml:"title" json:"title"`
	Status    Status      `yaml:"status" json:"status"`
	Owner     *string     `yaml:"owner,omitempty" json:"owner,omitempty"`
	EpicRefs  []EntityRef `yaml:"epic_refs,omitempty" json:"epic_refs,omitempty"`
	CreatedAt time.Time   `yaml:"created_at" json:"created_at"`
	UpdatedAt time.Time   `yaml:"updated_at" json:"updated_at"`
}

type Prd struct {
	PrdFrontmatter `yaml:",inline"`
	Body           string `yaml:"-" json:"body"`
	FilePath       string `yaml:"-" json:"file_path"`
}
